react-front-end/tsrc/search/components/DisplayModeSelector.tsx

```diff
@@ -42,48 +42,67 @@ export interface DisplayModeSelectorProps {
    * @param value the new Display Mode
    */
   onChange: (value: DisplayMode) => void;
+  /**
+   * `true` to disable Image mode.
+   */
+  disableImageMode: boolean;
+  /**
+   * `true` to disable Video mode.
+   */
+  disableVideoMode: boolean;
 }
 
 /**
  * A group of toggle buttons to provide a simple means of change the mode the search results are
  * displayed in.
  */
-const DisplayModeSelector = ({ value, onChange }: DisplayModeSelectorProps) => {
+const DisplayModeSelector = ({
+  value,
+  onChange,
+  disableImageMode,
+  disableVideoMode,
+}: DisplayModeSelectorProps) => {
   const options: {
     displayMode: DisplayMode;
     icon: ReactElement;
     label: string;
+    disabled: boolean;
   }[] = [
     {
       displayMode: "list",
       icon: <LibraryBooksIcon />,
       label: labelItemList,
+      disabled: false,
     },
     {
       displayMode: "gallery-image",
       icon: <PhotoLibraryIcon />,
       label: labelImageGallery,
+      disabled: disableImageMode,
     },
     {
       displayMode: "gallery-video",
       icon: <VideoLibraryIcon />,
       label: labelVideoGallery,
+      disabled: disableVideoMode,
     },
   ];
 
-  const buttons = options.map(({ displayMode, icon, label }) => {
+  const buttons = options.map(({ displayMode, icon, label, disabled }) => {
```

Comment:
Rather than making the JSX block more complicated, why not just:

```suggestion
  const buttons = options.filter(option => !option.disabled).map(({ displayMode, icon, label }) => {
```

Then all the JSX can stay how it used to be.